### ISSUE TITLE
SAE shape-uncertainty: dispersion phi-hat uses an approximate coordinate edf

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#676]
`SaeManifoldTerm::reconstruction_dispersion` (src/terms/sae_manifold.rs) computes the Gaussian reconstruction scale as `phi = RSS / (N*p - edf)` with `edf = (beta_dim - smooth_edf) + sum_i sum_k a_{ik}*d_k`.

The decoder term is exact (smooth_edf is the cached Fellner-Schall trace `tr(lambda*S_beta^-1*(+)S_k(x)I_p)`), but the **coordinate term over-counts**: it charges each soft assignment its full latent dim d_k and ignores the ARD shrinkage that reduces each coordinate's effective dof. So phi-hat is mildly conservative (bands a touch wide).

At SAE scale (N*p >> edf) the correction is negligible, but for calibrated **absolute** coverage on small problems the coordinate edf should be the exact ARD-shrunk trace, e.g. `sum_kj (1 - alpha_kj * tr_kj(H^-1))`, reusing the `ard_inverse_traces` already computed for the EFS step. This only rescales phi-hat globally; relative band widths (the trust-score property) are unaffected.

Powers the per-atom `decoder_covariance = phi * S_beta^-1[block]` and `shape_band` exposed on ManifoldSAE.

**Comment by SauersML** [comment]
Fixed in 880b96a91 (on `main`).

**Root cause.** `SaeManifoldTerm::reconstruction_dispersion` (src/terms/sae_manifold.rs) computed the latent-coordinate contribution to the residual dof as the assignment-weighted dimension budget `Σ_i Σ_k a_ik·d_k`. That charges each soft assignment its full latent dim `d_k` and ignores the ARD shrinkage that reduces each per-token coordinate's effective dof, so `φ̂ = RSS/(N·p − edf)` was mildly conservative (shape bands a touch wide). The decoder β term was already exact (cached Fellner-Schall smoothness trace).

**Fix.** The coordinate edf is now the exact MacKay/Fellner-Schall ARD-shrunk trace, per axis `edf_kj = n_active_k − α_kj·tr_kj(H⁻¹)`, with `α_kj = exp(ρ.log_ard[k][j])`. It **reuses the existing `ard_inverse_traces`** cache assembled for the EFS ARD step — no second, inconsistent trace is introduced (the function now takes `&SaeManifoldRho` instead of just `lambda_smooth`).

The subtlety the issue text's shorthand `Σ_kj (1 − α·tr)` glosses: `ard_inverse_traces` already *sums the posterior-variance over rows*, so the count term is not `1` per axis but the number of latent scalars on that axis, and it must match the trace's support exactly:
- dense layout (`last_row_layout == None`): trace sums all `n` rows ⇒ `n_active_k = n`;
- compact active-set layout (`Some`): trace sums only rows where atom `k` is active ⇒ `n_active_k = #{rows : k ∈ active_atoms}`.

An inactive (prior-dominated) coordinate contributes 0 to both the trace and the count, hence 0 edf — which is what a fully ARD-shrunk axis should cost. `edf_kj` is clamped to `[0, n_active_k]` against numerical drift.

**Convention consistency (verified against the file).** This is exactly the count/denominator pairing of the EFS ARD precision update in `efs_step`: `α_new = n_obs/(‖t‖² + tr(H⁻¹))` — dense `n` scalars per axis, the same `tr_kj(H⁻¹)` from `ard_inverse_traces`. The dense prior `ard_value` likewise sums `‖t‖²` over all `n` rows with a `−½·n·log α` term, confirming `n` (not soft mass) is the matched scalar count. Construction enforces `atom.latent_dim == coords[k].latent_dim()` (line ~2668), so `traces[k].len()` aligns with the axis loop; a shape guard returns `Err` otherwise.

As noted in the issue, this only globally rescales `φ̂`; relative band widths (the trust-score property) are unaffected. It powers the per-atom `decoder_covariance = φ·S_β⁻¹[block]` and `shape_band` on ManifoldSAE.

Honest note: not built/tested locally (disk near-full in this environment); reasoned from source and cross-checked with codex against the EFS edf convention. CI verifies.

**Comment by SauersML** [comment]
**Status: implemented** (landed in `880b96a91` "fix(sae): exact ARD-shrunk coordinate edf in reconstruction dispersion #676").

`SaeManifoldTerm::reconstruction_dispersion` (`src/terms/sae_manifold.rs:5023`) no longer charges each soft assignment its full latent dim. The coordinate edf is now the **exact ARD-shrunk trace**

```
coord_edf = Σ_k Σ_j ( n_active_k − α_{kj} · tr_{kj}(H⁻¹) ),   edf_kj ∈ [0, n_active_k]
```

- `tr_{kj}(H⁻¹)` is read from `ard_inverse_traces(cache)` (`:4886`) — the **same** per-coordinate-axis posterior-variance trace the EFS/Fellner–Schall ARD step uses (`α_new = n/(‖t‖²+tr(H⁻¹))`). It is **reused, not recomputed** (no extra inverse).
- `α_{kj} = exp(rho.log_ard[k][j])`.
- `n_active_k` is matched to the trace's support: `n` for the dense full-support layout, or the count of rows where atom `k` is active for the compact active-set layout — so a prior-dominated/inactive axis contributes 0 to both the trace and the count (0 edf).
- φ̂ = RSS / max(N·p − beta_edf − coord_edf, 1); `beta_edf = beta_dim − decoder_smoothness_effective_dof` (the exact cached Fellner–Schall smoothness trace) is unchanged.

This only rescales φ̂ globally; relative band widths (the trust-score property) are unaffected. Powers per-atom `decoder_covariance = φ·S_β⁻¹[block]` and `shape_band`.

Verify: `tests/test_sae_manifold_shape_uncertainty.py`. Coordinator should build + run before closing.

**Comment by SauersML** [comment]
Verified fixed and pushed the remaining regression-test repair.

Evidence:
- Production fix: `880b96a91` changed `SaeManifoldTerm::reconstruction_dispersion` to use the ARD-shrunk coordinate EDF via `ard_inverse_traces`.
- Regression fixture repair: `f2c4a1d73` changes the direct Rust test to use the correct K=1 softmax layout (`(n, 1)` logits).
- MSI verification on project-local caches/OpenBLAS:
  - `cargo test -q reconstruction_dispersion_uses_ard_shrunk_coordinate_edf --lib` -> `1 passed`
  - #677 MSI run also verified the shape uncertainty Python surface: `pytest -q tests/test_sae_manifold_shape_uncertainty.py tests/test_sae_manifold_public_api_shapes.py --tb=short` -> `5 passed`

The issue’s requested phi-hat EDF correction is covered and verified.
